app/assets/javascripts/admin/addon/components/admin-flags-form.gjs

```diff
@@ -0,0 +1,186 @@
+import { isEmpty } from "@ember/utils";
+import { not } from "truth-helpers";
+import DButton from "discourse/components/d-button";
+import { ajax } from "discourse/lib/ajax";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+import dIcon from "discourse-common/helpers/d-icon";
+import i18n from "discourse-common/helpers/i18n";
+import { bind } from "discourse-common/utils/decorators";
+import I18n from "discourse-i18n";
+import AdminConfigAreaCard from "admin/components/admin-config-area-card";
+import MultiSelect from "select-kit/components/multi-select";
+
+export default class AdminFlagsForm extends Component {
+  @service router;
+  @service site;
+
+  @tracked enabled = true;
+  @tracked name;
+  @tracked description;
+  @tracked appliesTo;
+
+  constructor() {
+    super(...arguments);
+    if (this.isUpdate) {
+      this.name = this.args.flag.name;
+      this.description = this.args.flag.description;
+      this.appliesTo = this.args.flag.applies_to;
+      this.enabled = this.args.flag.enabled;
+    }
+  }
+
+  get isUpdate() {
+    return this.args.flag;
+  }
+
+  get isValid() {
+    return (
+      !isEmpty(this.name) &&
+      !isEmpty(this.description) &&
+      !isEmpty(this.appliesTo)
+    );
+  }
+
+  get header() {
+    return this.isUpdate
+      ? "admin.config_areas.flags.form.edit_header"
+      : "admin.config_areas.flags.form.add_header";
+  }
+
+  get appliesToValues() {
+    return [
+      { name: I18n.t("admin.config_areas.flags.form.posts"), id: "Post" },
+      { name: I18n.t("admin.config_areas.flags.form.topics"), id: "Topic" },
+      {
+        name: I18n.t("admin.config_areas.flags.form.chat_messages"),
+        id: "Chat::Message",
+      },
+    ];
+  }
+
+  @action
+  save() {
+    this.isUpdate ? this.update() : this.create();
+  }
+
+  @bind
+  create() {
+    return ajax(`/admin/config/flags`, {
+      type: "POST",
+      data: this.#formData,
+    })
+      .then((flag) => {
+        this.site.flagTypes.push(flag);
+        this.router.transitionTo("adminConfig.flags");
+      })
+      .catch((error) => {
+        return popupAjaxError(error);
+      });
+  }
+
+  @bind
+  update() {
+    return ajax(`/admin/config/flags/${this.args.flag.id}`, {
+      type: "PUT",
+      data: this.#formData,
+    })
+      .then((flag) => {
+        this.args.flag.name = flag.name;
+        this.args.flag.description = flag.description;
+        this.args.flag.applies_to = flag.applies_to;
+        this.args.flag.enabled = flag.enabled;
+        this.router.transitionTo("adminConfig.flags");
+      })
+      .catch((error) => {
+        return popupAjaxError(error);
+      });
+  }
+
+  @bind
+  get #formData() {
+    return {
+      name: this.name,
+      description: this.description,
+      applies_to: this.appliesTo,
+      enabled: this.enabled,
+    };
+  }
+
+  <template>
+    <div class="admin-config-area">
+      <h2>{{i18n "admin.config_areas.flags.header"}}</h2>
+      <LinkTo
```

Comment:
We have a `BackButton` component for this:

https://github.com/discourse/discourse/blob/5cd58e7aab15414b588cd53348a6c14e55c08e93/app/assets/javascripts/discourse/app/components/back-button.gjs

Example from AI:

https://github.com/discourse/discourse-ai/blob/fba8673ed0309f6f80a1a8ba466593f4561bbccc/assets/javascripts/discourse/components/ai-llm-editor.gjs#L114-L117